Resources/public/js/views/services/plugins/ez-universaldiscoverycontentsearchplugin.js

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (C) eZ Systems AS. All rights reserved.
+ * For full copyright and license information view LICENSE file distributed with this source code.
+ */
+YUI.add('ez-universaldiscoverycontentsearchplugin', function (Y) {
+    "use strict";
+    /**
+     * Provides the content search plugin for the universal discovery
+     *
+     * @module ez-universaldiscoverycontentsearchplugin
+     */
+    Y.namespace('eZ.Plugin');
+
+    /**
+     * Universal Discovery Content Search Plugin. It enhances the universal
+     * discovery to handle the content search related events and fetching.
+     *
+     * @namespace eZ.Plugin
+     * @class UniversalDiscoveryContentSearch
+     * @constructor
+     */
+    Y.eZ.Plugin.UniversalDiscoveryContentSearch = Y.Base.create('universalDiscoveryContentSearch', Y.eZ.Plugin.ViewServiceBase, [], {
+        initializer: function () {
+            this.onHostEvent('universalDiscoverySearchView:searchContent', this._search);
+        },
+
+        _search: function (e) {
+            var capi = this.get('host').get('capi'),
+                contentService = capi.getContentService(),
+                searchText = e.searchText,
+                query = contentService.newViewCreateStruct('udw-content-search-' + searchText),
```

Comment:
you should do a Location search instead of a Content search because this way you get both the Location and ContentInfo that are needed for the UDW selection handling. But actually, you should reuse the locationSearchPlugin, it does pretty much the same thing (and already support the pagination).

About the needed objects, you'll also need the content type. I think you should add a `loadContentType` option to the `locationSearch` event parameter so that the plugin also load the content type of each result in addition to searching.